bgpd/bgp_main.c

```diff
@@ -294,19 +294,6 @@ static int bgp_vrf_enable(struct vrf *vrf)
 
 	bgp = bgp_lookup_by_name(vrf->name);
 	if (bgp && bgp->vrf_id != vrf->vrf_id) {
-		if (bgp->name && strmatch(vrf->name, VRF_DEFAULT_NAME)) {
-			XFREE(MTYPE_BGP, bgp->name);
-			XFREE(MTYPE_BGP, bgp->name_pretty);
-			bgp->name_pretty = XSTRDUP(MTYPE_BGP, "VRF default");
-			bgp->inst_type = BGP_INSTANCE_TYPE_DEFAULT;
-#ifdef ENABLE_BGP_VNC
-			if (!bgp->rfapi) {
-				bgp->rfapi = bgp_rfapi_new(bgp);
-				assert(bgp->rfapi);
-				assert(bgp->rfapi_cfg);
-			}
-#endif /* ENABLE_BGP_VNC */
-		}
```

Comment:
Perhaps I'm misreading the original code, but it looks like it was there to handle the case where a VRF is configured first and then the default VRF is added.  How does the modified code cover this case?

Reply:
No, this code is needed to handle the following situation:
- bgpd starts and reads the config with `router bgp X vrf Y` - and treats this as VRF config
- bgpd connects to zebra and learns that `Y` is actually the name of the default VRF (we allow that currently)
- bgpd needs to change the instance type to `DEFAULT` and fixup the names

The ability to rename the default VRF after the daemon is started is not compatible with the mgmtd model. So in this PR I remove this ability and allow the default VRF to be renamed only at the startup using the `-o` option. Hence this code is not needed anymore (and similar code in all other daemons).

But there's always a possibility that I missed something and the CI is failing because of that.